### PR TITLE
Callback plugins: show type, render indexes of callback plugins per type

### LINKED ISSUE #89: Show callback type
Right now the type of a callback plugin isn't shown in the HTML docsite, while it is shown by ansible-doc (ref: https://github.com/ansible-collections/community.general/issues/5734#issuecomment-1365478929).

### PR DESCRIPTION
**Pull request by felixfontein**
Fixes #89.

Examples:
- callback plugin index: https://ansible.fontein.de/collections/index_callback.html - see links to specialized indexes at top:
  - https://ansible.fontein.de/collections/callback_index_aggregate.html
  - https://ansible.fontein.de/collections/callback_index_notification.html
  - https://ansible.fontein.de/collections/callback_index_stdout.html
- default callback (stdout): https://ansible.fontein.de/collections/ansible/builtin/default_callback.html#callback-plugin
- splunk callback (aggregate): https://ansible.fontein.de/collections/community/general/splunk_callback.html#callback-plugin
- say callback (notification): https://ansible.fontein.de/collections/community/general/say_callback.html#callback-plugin

### CODE REVIEW

**Review by oraNod — COMMENTED** [review]
LGTM

File: tests/functional/baseline-no-breadcrumbs/collections/index_callback.rst [hunk 1]
```diff
@@ -6,6 +6,14 @@
 Index of all Callback Plugins
 =============================
 
+.. toctree::
+   :maxdepth: 1
+   :caption: List of callback plugin by callback type
```
Comment (samccann) [inline tests/functional/baseline-no-breadcrumbs/collections/index_callback.rst:11]:
```suggestion
   :caption: List of callback plugins by callback type
```

File: tests/functional/baseline-use-html-blobs/collections/index_callback.rst [hunk 2]
```diff
@@ -6,6 +6,14 @@
 Index of all Callback Plugins
 =============================
 
+.. toctree::
+   :maxdepth: 1
+   :caption: List of callback plugin by callback type
```
Comment (samccann) [inline tests/functional/baseline-use-html-blobs/collections/index_callback.rst:11]:
```suggestion
   :caption: List of callback plugins by callback type
```

**Review by samccann — COMMENTED** [review]
I probably didn't need to change the nits in the test files but I was half-way thru before I realized I was... in test files ;-)

**Comment by felixfontein** [comment]
CI failure is unrelated.

**Comment by felixfontein** [comment]
@oraNod @samccann thanks for reviewing this!

**Pull request merged**